src/test/run-make-fulldeps/issue-64319/Makefile

```diff
@@ -0,0 +1,10 @@
+-include ../tools.mk
+
+all:
+	$(RUSTC) --crate-type rlib foo.rs
+	$(RUSTC) --crate-type dylib bar.rs
+	$(RUSTC) --crate-type dylib bar.rs -C opt-level=1
+	$(RUSTC) --crate-type dylib bar.rs -C opt-level=2
+	$(RUSTC) --crate-type dylib bar.rs -C opt-level=3
+	$(RUSTC) --crate-type dylib bar.rs -C opt-level=s
+	$(RUSTC) --crate-type dylib bar.rs -C opt-level=z
```

Comment:
Could you also add `$(RUSTC) --crate-type dylib bar.rs -C opt-level=3 -Zshare-generics=on` ?

Reply:
Also, could you please add a comment saying that different optimization levels imply different settings for generics sharing?